### PR TITLE
ID-1082 Initial Commit

### PR DESCRIPTION
**Pull request by tlangs**
Setting up Cloud Transparency Manager with a bit more than the default skeleton. Still a lot more work to be done, but might as well save progress now that tests pass.

### CODE REVIEW

**Review by mikeflinn — APPROVED** [review]
Approving for the sake of getting a first image built and pushed

File: service/src/main/java/bio/terra/cloudtransparency/dao/CloudTransparentAccessDao.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  @WithSpan
+  public void upsertCloudTransparentAccess(CloudTransparentAccess cloudTransparentAccess) {
+    var query =
+        "INSERT INTO cloud_transparent_access (sam_user_id, resource_type_name, resource_id, enabled, updated_at)"
+            + " VALUES (:samUserId, :resourceTypeName, :resourceId, :enabled, current_timestamp)"
+            + " ON CONFLICT (sam_user_id, resource_type_name, resource_id) DO UPDATE SET"
+            + " enabled = excluded.enabled, "
+            + " updated_at = current_timestamp";
```
Comment (Ghost-in-a-Jar) [inline service/src/main/java/bio/terra/cloudtransparency/dao/CloudTransparentAccessDao.java:37]: love me a good ol' upsert

File: service/src/main/resources/application.yml [hunk 2]
```diff
@@ -6,13 +6,23 @@ env:
...
     samplingRate: ${SAMPLING_PROBABILITY:0}
   sam:
     basePath: ${SAM_ADDRESS:https://sam.dsde-dev.broadinstitute.org}
+  googleDirectoryApi:
+    apiServiceAccounts:
+      apiServiceAccountPath1: ${GOOGLE_DIRECTORY_API_SERVICE_ACCOUNT_1:/service/src/main/resources/rendered/admin-service-account-1.json}
+      apiServiceAccountPath2: ${GOOGLE_DIRECTORY_API_SERVICE_ACCOUNT_2:/service/src/main/resources/rendered/admin-service-account-2.json}
+      apiServiceAccountPath3: ${GOOGLE_DIRECTORY_API_SERVICE_ACCOUNT_3:/service/src/main/resources/rendered/admin-service-account-3.json}
+      apiServiceAccountPath4: ${GOOGLE_DIRECTORY_API_SERVICE_ACCOUNT_4:/service/src/main/resources/rendered/admin-service-account-4.json}
+      apiServiceAccountPath5: ${GOOGLE_DIRECTORY_API_SERVICE_ACCOUNT_5:/service/src/main/resources/rendered/admin-service-account-5.json}
```
Comment (Ghost-in-a-Jar) [inline service/src/main/resources/application.yml:22]: kinda hilarious we have to do this

File: service/src/test/java/bio/terra/cloudtransparency/service/CloudTransparentAccessServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            .resourceId("resourceId")
+            .enabled(true);
+    when(cloudTransparentAccessDao.getCloudTransparentAccessesForUser(any()))
+        .thenReturn(List.of(testAccessRecord));
+
+    var result =
+        cloudTransparentAccessService.getCloudTransparentAccessesForUser(
+            expectedAccess.getSamUserId());
+    assertEquals(List.of(expectedAccess), result);
+  }
```
Comment (Ghost-in-a-Jar) [inline service/src/test/java/bio/terra/cloudtransparency/service/CloudTransparentAccessServiceTest.java:67]: maybe add a delete test case here too?

**Review by Ghost-in-a-Jar — APPROVED** [review]
some minor comments, looks good over all

File: service/src/test/java/bio/terra/cloudtransparency/service/CloudTransparentAccessServiceTest.java (hunk 3 above)
Comment (tlangs) [reply service/src/test/java/bio/terra/cloudtransparency/service/CloudTransparentAccessServiceTest.java:67]: There's no delete method in the dao, so there's no delete test case!

**Pull request merged**